Compression/Compression.cs

```diff
@@ -524,6 +527,91 @@ public static StreamReader Unzip(string filename, string zipEntryName, out ZipFi
             return reader;
         }
 
+
+        // ZipArchive cache used by UnzipCached method
+        private static ConcurrentDictionary<string, ZipArchive> _zipArchiveCache = new ConcurrentDictionary<string, ZipArchive>();
```

Comment:
Style/Design - Caching of zip archives, if possible given current design, may be better suited to instance instead of static state.

Static state can get hairy, leads to concurrency bugs, and tight coupling of components with implementations.